pkg/operator/controller/canary/route_test.go

```diff
@@ -134,10 +134,13 @@ func TestCanaryRouteChanged(t *testing.T) {
 		mutated := original.DeepCopy()
 		tc.mutate(mutated)
 		if changed, updated := canaryRouteChanged(original, mutated); changed != tc.expect {
-			t.Errorf("%s, expect canaryRouteChanged to be %t, got %t", tc.description, tc.expect, changed)
+			t.Errorf("expected canaryRouteChanged to be %t, got %t", tc.expect, changed)
 		} else if changed {
+			if updatedChanged, _ := canaryRouteChanged(original, updated); !updatedChanged {
+				t.Error("canaryRouteChanged reported changes but did not make any update")
+			}
 			if changedAgain, _ := canaryRouteChanged(mutated, updated); changedAgain {
-				t.Errorf("%s, canaryRouteChanged does not behave as a fixed point function", tc.description)
+				t.Error("canaryRouteChanged does not behave as a fixed point function")
```

Comment:
Here, `t.Run` _isn't_ added, but the formats are changed to omit `tc.description`.  Very confusing.